pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -84,67 +85,68 @@ func TestDevicePluginReRegistration(t *testing.T) {
 	devsForRegistration := []*pluginapi.Device{
 		{ID: "Dev3", Health: pluginapi.Healthy},
 	}
+	for _, preStartFlag := range []bool{false, true} {
 
-	expCallbackCount := int32(0)
-	callbackCount := int32(0)
-	callbackChan := make(chan int32)
-	callback := func(n string, a, u, r []pluginapi.Device) {
-		callbackCount++
-		if callbackCount > atomic.LoadInt32(&expCallbackCount) {
+		expCallbackCount := int32(0)
+		callbackCount := int32(0)
+		callbackChan := make(chan int32)
+		callback := func(n string, a, u, r []pluginapi.Device) {
+			callbackCount++
+			if callbackCount > atomic.LoadInt32(&expCallbackCount) {
+				t.FailNow()
+			}
+			callbackChan <- callbackCount
+		}
+		m, p1 := setup(t, devs, callback, socketName, pluginSocketName)
+		atomic.StoreInt32(&expCallbackCount, 1)
+		p1.Register(socketName, testResourceName, preStartFlag)
+		// Wait for the first callback to be issued.
+
+		select {
+		case <-callbackChan:
+			break
+		case <-time.After(time.Second):
+			t.FailNow()
```

Comment:
This part is not being newly added in this PR. Such improvements we can do in individual PRs.

Reply:
Execpt that the 5 FailNow() (4 existing one added) of this file are being affected by this PR.
Can you change them?

Reply:
I can count 4 FailNow() in existing file:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/devicemanager/manager_test.go

Additionally, I would prefer test cases to continue to run and get a list of all unit test failures at end in one iteration.

Reply:
> I can count 4 FailNow() in existing file:

Because you are adding the 5th one.

> Additionally, I would prefer test cases to continue to run and get a list of all unit test failures at end in one iteration.

The point of the FailNow() here is that the test is stuck waiting for a channel. If you don't FailNow your test will never finish.
The problem we have right now witht FailNow() is that it doesn't print a backtrace or an error it just exits the test. Can you replace the FailNow() your PR is affecting by a t.Fatal()?